abci/types/application.go

```diff
@@ -14,7 +14,8 @@ type Application interface {
 	Query(RequestQuery) ResponseQuery // Query for state
 
 	// Mempool Connection
-	CheckTx(RequestCheckTx) ResponseCheckTx // Validate a tx for the mempool
+	CheckTx(RequestCheckTx) ResponseCheckTx                   // Validate a tx for the mempool
+	PreprocessTxs(RequestPreprocessTxs) ResponsePreprocessTxs // State machine preprocessing of txs
```

Comment:
Not sure if "Mempool Connection" is the right 'section'. Don't you think "Consensus Connection" would be the right place? That does not matter much though.

Reply:
It's already on the consensus connection just not in the right place in the interface.